### PR TITLE
fix: automatically install using make if flock isn't installed. and use bashsource to overcome shebang

### PR DESCRIPTION
**Pull request by maxwelllwang**
@bukzor

### CODE REVIEW

File: activate.sh [hunk 1]
```diff
@@ -1,7 +1,14 @@
 #!/sourceme/bash
-_here="$(readlink -f "$(dirname "$0")")"
+_here="$(readlink -f "$(dirname "${BASH_SOURCE:-$0}")")"
 
 export TACOS_GHA_HOME="$_here"
+
+if ! command -v flock &> /dev/null; then
+    echo "flock is not installed. Running make..."
+    make -C "$TACOS_GHA_HOME" .done/brew
+fi
```
Comment (bukzor) [inline activate.sh:9]:
```suggestion
    echo "Installing missing dependencies... (flock)"
```

File: activate.sh [hunk 2]
```diff
@@ -1,7 +1,14 @@
 #!/sourceme/bash
-_here="$(readlink -f "$(dirname "$0")")"
+_here="$(readlink -f "$(dirname "${BASH_SOURCE:-$0}")")"
```
Comment (bukzor) [inline activate.sh:2]: Please test: if you source this from your interactive zsh shell, it works well?


```
$ cd ops
$ which sudo-gcp  # should fail
$ (set -x; source ../tacos-gha/activate.sh; which sudo-gcp)
```

**Review by bukzor — CHANGES_REQUESTED** [review]

File: activate.sh (hunk 2 above)
Comment (maxwelllwang) [reply activate.sh:2]:
```
+/bin/zsh:12> source ../tacos-gha/activate.sh
+../tacos-gha/activate.sh:2> _here=+../tacos-gha/activate.sh:2> dirname ../tacos-gha/activate.sh
+../tacos-gha/activate.sh:2> _here=+../tacos-gha/activate.sh:2> readlink -f ../tacos-gha
+../tacos-gha/activate.sh:2> _here=/Users/maxwelllwang/code/tacos-gha 
+../tacos-gha/activate.sh:4> export TACOS_GHA_HOME=/Users/maxwelllwang/code/tacos-gha
+../tacos-gha/activate.sh:6> command -v flock
+../tacos-gha/activate.sh:7> echo 'Installing missing dependencies... (flock)'
Installing missing dependencies... (flock)
+../tacos-gha/activate.sh:8> make -C /Users/maxwelllwang/code/tacos-gha .done/brew
make: `.done/brew' is up to date.
+../tacos-gha/activate.sh:11> export PATH='/Users/maxwelllwang/code/tacos-gha/bin:/Users/maxwelllwang/.nvm/versions/node/v19.9.0/bin:/opt/homebrew/opt/openjdk/bin:/Users/maxwelllwang/.pyenv/shims:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/salt:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/maxwelllwang/.nvm/versions/node/v19.9.0/bin:/opt/homebrew/opt/openjdk/bin:/Users/maxwelllwang/.volta/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/opt/homebrew/opt/fzf/bin}'
+../tacos-gha/activate.sh:14> unset _here
+/bin/zsh:12> which sudo-gcp
/Users/maxwelllwang/code/tacos-gha/bin/sudo-gcp
```

**Comment by maxwelllwang** [comment]
new PR

**Pull request closed without merging**